pkg/querier/querier.go

```diff
@@ -97,30 +100,36 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxQueryLookback time.Duration) storage.Queryable {
+func NewQueryable(dq, cq storage.Queryable, distributor Distributor, cfg Config) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-		cqr, err := cq.Querier(ctx, mint, maxt)
-		if err != nil {
-			return nil, err
-		}
-
 		q := querier{
-			queriers:    []storage.Querier{cqr},
 			distributor: distributor,
 			ctx:         ctx,
 			mint:        mint,
 			maxt:        maxt,
 		}
 
 		// Include ingester only if maxt is within ingesterMaxQueryLookback w.r.t. current time.
-		if ingesterMaxQueryLookback == 0 || maxt >= time.Now().Add(-ingesterMaxQueryLookback).UnixNano()/1e6 {
+		if cfg.IngesterMaxQueryLookback == 0 || maxt >= time.Now().Add(-cfg.IngesterMaxQueryLookback).UnixNano()/1e6 {
 			dqr, err := dq.Querier(ctx, mint, maxt)
 			if err != nil {
 				return nil, err
 			}
 			q.queriers = append(q.queriers, dqr)
+
+			// query fits within IngesterMaxQueryLookback we can skip sending queries to LTS
```

Comment:
I guess I'm missing something. How can you assert the query fits within `IngesterMaxQueryLookback`? We have just checked `maxt` but not `mint`.

_Same comment applies to `newUnifiedChunkQueryable()`_

Reply:
Fixed. I've moved the MinChunkAge logic into the querier as IngesterMetricMinAge